snuba/datasets/__init__.py

```diff
@@ -1,12 +1,33 @@
-from typing import Optional, Mapping, Sequence, Tuple
+from typing import (
+    Any,
+    Optional,
+    Mapping,
+    Sequence,
+    Tuple,
+)
+from typing_extensions import Protocol
 
 from snuba.datasets.dataset_schemas import DatasetSchemas
 from snuba.datasets.table_storage import TableWriter
 from snuba.query.extensions import QueryExtension
+from snuba.request import Request
 
 from snuba.util import escape_col, parse_datetime
 
 
+class SplitQueryFunc(Protocol):
+    """
+    The function definition for parse_and_run_query. Used for typing.
+    """
+    def __call__(
+            self,
+            dataset: 'Dataset',
```

Comment:
This is a forward ref: https://mypy.readthedocs.io/en/latest/kinds_of_types.html#class-name-forward-references. Haven't seen it used in the repo before so figured I'd put down a link.

Reply:
Add 
```
from __futures__ import annotations
```
instead.
https://www.python.org/dev/peps/pep-0563/
So you do not have to define the type as string.